docs/dev.md

```diff
@@ -1,21 +1,33 @@
-## Development & Deploy
+## Development
 
-Install ytt, kbld, kapp beforehand (https://k14s.io).
+### Prerequisites
 
+Install ytt, kbld, kapp beforehand (https://k14s.io). We provide a handy script to pull the required dependencies.
+
+```
+./hack/install-deps.sh
 ```
-./hack/build.sh # to build locally
+### Build Locally
 
-# deploys secretgen-controller with kapp-controller
-# and also runs tests where kapp-controller integrates 
-# with secretgen-controller
-export KAPPCTRL_E2E_SECRETGEN_CONTROLLER=true
+```
+./hack/build.sh
+```
+
+### Test e2e
 
-# add `-v image_repo=docker.io/username/kapp-controller` with your registry to ytt invocation inside
-./hack/deploy.sh # to deploy
```

Comment:
I replaced `./hack/deploy.sh` with `./hack/deploy-test.sh` - It appears that deploy-test is a superset of deploy and just adds a few more resources to the cluster for testing, for this doc it felt unnecessary to mention both.